invenio/modules/oaiharvester/tasks/postprocess.py

```diff
@@ -297,53 +243,49 @@ def refextract(obj, eng):
     :param eng: BibWorkflowEngine processing the object
     """
     from invenio.legacy.refextract.api import extract_references_from_file_xml
-    from invenio.utils.plotextractor.getter import harvest_single
+    from invenio.utils.plotextractor.api import get_pdf_from_arxiv
     from invenio.modules.workflows.utils import convert_marcxml_to_bibfield
 
     if "_result" not in obj.extra_data:
         obj.extra_data["_result"] = {}
 
-    pdf = None
-
-    if "_result" in obj.extra_data and "pdf" in obj.extra_data["_result"]:
+    try:
         pdf = obj.extra_data["_result"]["pdf"]
+    except KeyError:
+        pdf = None
 
     if not pdf:
         extract_path = os.path.join(
-            cfg['CFG_TMPSHAREDDIR'],
+            cfg.get('OAIHARVESTER_STORAGEDIR', cfg.get('CFG_TMPSHAREDDIR')),
             str(eng.uuid)
         )
-        if not os.path.exists(extract_path):
-            os.makedirs(extract_path)
-        tarball, pdf = harvest_single(
-            obj.data["system_control_number"]["value"], extract_path, ["pdf"]
+        pdf = get_pdf_from_arxiv(
+            obj.data.get(cfg.get('OAIHARVESTER_RECORD_ARXIV_ID_LOOKUP')),
+            extract_path
         )
         obj.extra_data["_result"]["pdf"] = pdf
 
-    if pdf and os.path.isfile(obj.extra_data["_result"]["pdf"]):
+    if pdf and os.path.isfile(pdf):
         references_xml = extract_references_from_file_xml(
-            obj.extra_data["_result"]["pdf"])
+            obj.extra_data["_result"]["pdf"]
+        )
         if references_xml:
-            obj.log.info("Found references: {0}".format(references_xml))
             updated_xml = '<?xml version="1.0" encoding="UTF-8"?>\n' \
                           '<collection>\n' + references_xml + \
                           "\n</collection>"
-
             new_dict_representation = convert_marcxml_to_bibfield(updated_xml)
-            try:
-                obj.data['reference'].append(
-                    new_dict_representation["reference"])
-            except KeyError:
-                if 'reference' in new_dict_representation:
-                    obj.data['reference'] = [
-                        new_dict_representation['reference']]
-            obj.add_task_result("References",
-                                new_dict_representation['reference'],
-                                "workflows/results/refextract.html")
+            obj.data["reference"] = new_dict_representation["reference"]
+            obj.log.info("Extracted {0} references".format(len(obj.data["reference"])))
+            obj.update_task_results(
+                "References",
+                [{"name": "References",
+                  "result": new_dict_representation['reference'],
+                  "template": "workflows/results/refextract.html"}]
```

Comment:
We could, I guess. However, throughout Invenio 2 referencing jinja templates have usually been written this way everywhere. We should then change it everywhere which is probably out of scope for this update.